python/tvm/relay/frontend/pytorch.py

```diff
@@ -115,64 +115,70 @@ def inplace_add_to_add(op_name):
     return False
 
 
+
 # operator implementation
 def _elemwise(name):
     def _impl(inputs, input_types):
-        # TODO: Figure out a better way to get typing to work for tensor + scalar
-        type0 = input_types[0]
-        if isinstance(inputs[1], _expr.Expr):
-            type0 = input_types[1]
-
-        type1 = input_types[1]
-        if isinstance(inputs[0], _expr.Expr):
-            type1 = input_types[0]
-
-        data0 = _convert_elemwise_input(inputs[0], type0)
-        data1 = _convert_elemwise_input(inputs[1], type1)
-
+        data0, data1 = _pytorch_promote_types(inputs[:2], input_types[:2])
```

Comment:
i was trying to run gp2 model and i was getting some error with the new modifications. 
Code you can download from 
https://gist.github.com/siju-samuel/34e63e0719e06679b5c3688bce7a0515

Error is

```
Traceback (most recent call last):
  File "gp2.py", line 26, in <module>
    mod, params = relay.frontend.from_pytorch(scripted_model, input_shapes)
  File "/home/siju/workspace/tvm/python/tvm/relay/frontend/pytorch.py", line 2645, in from_pytorch
    ret = convert_operators(_get_operator_nodes(graph.nodes()),
  File "/home/siju/workspace/tvm/python/tvm/relay/frontend/pytorch.py", line 2555, in convert_operators
    relay_out = relay_op(inputs, _get_input_types(op_node, default_dtype=default_dtype))
  File "/home/siju/workspace/tvm/python/tvm/relay/frontend/pytorch.py", line 1694, in _impl
    return _elemwise("add")(inputs, input_types)
  File "/home/siju/workspace/tvm/python/tvm/relay/frontend/pytorch.py", line 151, in _impl
    return get_relay_op(name)(data0, data1)
  File "/home/siju/workspace/tvm/python/tvm/relay/op/tensor.py", line 513, in add
    return _make.add(lhs, rhs)
  File "/home/siju/workspace/tvm/python/tvm/_ffi/_ctypes/packed_func.py", line 225, in __call__
    raise get_last_ffi_error()
tvm._ffi.base.TVMError: Traceback (most recent call last):
  [bt] (4) /home/siju/workspace/tvm/build/libtvm.so(TVMFuncCall+0x69) [0x7f983831cc09]
  [bt] (3) /home/siju/workspace/tvm/build/libtvm.so(+0xa5be6b) [0x7f9837f31e6b]
  [bt] (2) /home/siju/workspace/tvm/build/libtvm.so(tvm::runtime::TVMMovableArgValue_::operator tvm::RelayExpr<tvm::RelayExpr, void>() const+0x63) [0x7f9837e0ecf3]
  [bt] (1) /home/siju/workspace/tvm/build/libtvm.so(tvm::RelayExpr tvm::runtime::TVMPODValue_::AsObjectRef<tvm::RelayExpr>() const+0x1a6) [0x7f9837ab21b6]
  [bt] (0) /home/siju/workspace/tvm/build/libtvm.so(+0x5cc26b) [0x7f9837aa226b]
  File "/home/siju/workspace/tvm/include/tvm/runtime/packed_func.h", line 1423
TVMError: Check failed: type_code_ == kTVMObjectHandle (0 vs. 8) : expected Object but get int
```

Reply:
I will look into this. Thank you!

Reply:
Can we standardise on transformers btw? it produces the same error, so...

Reply:
without this PR changes, that error was not there. May be some cases are missed.

Reply:
So the immediate error was caused by numpy scalars instead of Python number. :confused: 
That is easily fixed. I do bump into something about the shape in a where statement not being broadcast. That's probably unrelated to the dtype fixes, but I can throw in broadcasting for where (the question being whether we want it in relay, too, or just in the frontend).

Reply:
i think now you can throw error from frontend.

Reply:
So I fixed the where, but I then get errors in other obscure corners. Did the GPT2 model translate correctly with the old code?

Reply:
Other erros are there. Its ok. im still working on that.

Reply:
Ah, OK. :sweat_smile:   Well, do you want the broadcasting for where to go in this patch or should I just leave it.

Reply:
I would have a tendency to leave it (and also not raise in the frontend but leave _op.where to complain because broadcasting is substantial to test for) and then I can do a followup PR for where broadcasting.

Reply:
Its not related to this PR. So u can leave it for now and raise another followup PR to support broadcasting for `where`.

Reply:
OK. So I think I have the other bits covered.

Reply:
Thank you!